File: java-boot/src/test/java/se/callistaenterprise/kotlindsl/repo/CustomerRepositoryTest.java
package se.callistaenterprise.kotlindsl.repo;

import org.junit.jupiter.api.Test;
import org.springframework.beans.factory.annotation.Autowired;
import se.callistaenterprise.kotlindsl.model.CustomerIn;
import se.callistaenterprise.kotlindsl.model.CustomerStatus;
import se.callistaenterprise.kotlindsl.testutil.DslIntegrationTest;
import se.callistaenterprise.kotlindsl.testutil.DslJooqTest;

import static org.junit.jupiter.api.Assertions.assertEquals;

@DslJooqTest
public class CustomerRepositoryTest extends DslIntegrationTest {

    private final CustomerRepository customerRepo;

    CustomerRepositoryTest(@Autowired CustomerRepository customerRepo) {
        this.customerRepo = customerRepo;
    }

    @Test
    public void roundTripWorks() {
        var customerIn = new CustomerIn("1234", CustomerStatus.ACTIVE, "A", "Customer");
        var id = customerRepo.create(customerIn);
        var customer = customerRepo.getById(id);
        assertEquals(customer.customerId(), customerIn.customerId());
        assertEquals(customer.status(), customerIn.status());
        assertEquals(customer.firstName(), customerIn.firstName());
        assertEquals(customer.lastName(), customerIn.lastName());
    }

}

File: kotlin-boot/src/test/kotlin/se/callistaenterprise/kotlindsl/repo/CustomerRepositoryTest.kt
package se.callistaenterprise.kotlindsl.repo

import io.kotest.matchers.shouldBe
import org.junit.jupiter.api.Test
import org.springframework.beans.factory.annotation.Autowired
import se.callistaenterprise.kotlindsl.model.CustomerIn
import se.callistaenterprise.kotlindsl.model.CustomerStatus
import se.callistaenterprise.kotlindsl.testutil.DslIntegrationTest
import se.callistaenterprise.kotlindsl.testutil.DslJooqTest

@DslJooqTest
class CustomerRepositoryTest(@Autowired val customerRepo: CustomerRepository) : DslIntegrationTest() {

    @Test
    fun `Roundtrip works`() {
        val customerIn = CustomerIn("1234", CustomerStatus.ACTIVE, "A", "Customer")
        val id = customerRepo.create(customerIn)
        val customer = customerRepo.getById(id)!!
        customer.customerId shouldBe customerIn.customerId
        customer.status shouldBe customerIn.status
        customer.firstName shouldBe customerIn.firstName
        customer.lastName shouldBe customerIn.lastName
    }

}
